clouddq/classes/dq_rule_binding.py

```diff
@@ -334,11 +336,32 @@ def resolve_all_configs_to_dict(
             row_filter_config = self.resolve_row_filter_config(configs_cache)
             # resolve reference columns config
             if self.reference_columns_id:
+                include_all_reference_columns = False
                 include_reference_columns = self.resolve_reference_columns_config(
                     configs_cache
                 ).include_reference_columns
```

Comment:
this is obsolete if "*" is in include_reference_columns

Reply:
could we optimize the if condition clauses a bit more? there are duplicated and redundant work still happening